native/cocos/renderer/pipeline/RenderPipeline.cpp

```diff
@@ -124,6 +125,20 @@ void RenderPipeline::destroyQuadInputAssembler() {
     _quadIA.clear();
 }
 
+void RenderPipeline::updateGeometryRenderer(const ccstd::vector<scene::Camera *> &cameras) {
+    if (_geometryRenderer) {
+        return ;
+    }
+    
+    // Query the first camera rendering to swapchain.
+    for (const auto *camera : cameras) {
+        if (camera && camera->getWindow() && camera->getWindow()->getSwapchain() ) {
+            _geometryRenderer = camera->getGeometryRenderer();
+            return;
+        }
+    }
+}
+
 bool RenderPipeline::destroy() {
```

Comment:
Should set `_geometryRenderer ` to `nullptr` here?

Reply:
ok